src/client/js/Utils/LibraryManager.js

```diff
@@ -108,5 +126,170 @@ define([
 
     };
 
+    LibraryManager.prototype.check = function (name, callback) {
+        var client = this._client,
+            availableNames = client.getLibraryNames(),
+            libraryInfo,
+            notification;
+
+        if (availableNames.indexOf(name) !== -1) {
+            libraryInfo = client.getLibraryInfo(name);
+            if (libraryInfo && libraryInfo.projectId && libraryInfo.branchName) {
+                client.getBranches(libraryInfo.projectId, function (err, branches) {
+                    if (err) {
+                        callback(err);
+                    } else if (branches[libraryInfo.branchName] &&
+                        branches[libraryInfo.branchName] !== libraryInfo.commitHash) {
+                        client.notifyUser({message: 'New version available from [' + name + '] library'});
+                        callback(null, true);
+                    } else {
+                        callback(null, false);
+                    }
+                });
+            }
+        }
+    };
+
+    LibraryManager.prototype._clearWatches = function () {
+        var projectId;
+
+        for (projectId in this._followedProjects) {
+            this._client.unwatchProject(projectId, this._followedProjects[projectId]);
+        }
+        this._followedProjects = {};
+    };
+
+    LibraryManager.prototype._unwatchProject = function (projectId) {
+        var libraryName,
+            stillWatching = false;
+
+        for (libraryName in this._libraryInfos) {
+            if (this._libraryInfos[libraryName].projectId === projectId &&
+                this._libraryInfos[libraryName].notified === false) {
+                stillWatching = true;
+            }
+        }
+
+        if (stillWatching === false) {
+            this._client.unwatchProject(projectId, this._followedProjects[projectId]);
+            delete this._followedProjects[projectId];
+        }
+    };
+
+    LibraryManager.prototype._watchProject = function (projectId) {
+        var self = this,
+            eventFunction = function (websocket, event) {
+                var libraryName;
+
+                if (event && event.etype && event.etype === CONSTANTS.BRANCH_HASH_UPDATED &&
```

Comment:
Two first checks are redundant - an event is always returned and event.projectId too.